gcloud-java-core/src/main/java/com/google/gcloud/BatchResult.java

```diff
@@ -19,31 +19,64 @@
 /**
  * This class holds a single result of a batch call.
  */
-public interface BatchResult<T, E extends BaseServiceException> {
+public abstract class BatchResult<T, E extends BaseServiceException> {
+
+  private T result;
+  private boolean submitted = false;
+  private E error;
 
   /**
-   * Returns {@code true} if the batch has been submitted and the result is available; {@code
-   * false} otherwise.
+   * Returns {@code true} if the batch has been submitted and the result is available; {@code false}
+   * otherwise.
    */
-  boolean submitted();
+  public boolean submitted() {
+    return submitted;
+  }
 
   /**
    * Returns result of this call.
    *
-   * @throws IllegalArgumentException if the batch has not been submitted yet
+   * @throws IllegalStateException if the batch has not been submitted yet
    * @throws E if an error occurred when processing this request
    */
-  T get() throws E;
+  public T get() throws E {
+    if (!submitted()) {
+      throw new IllegalStateException("Batch has not been submitted yet");
```

Comment:
Nit: you could use [checkState(submitted(), "Batch has not been submitted yet")](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Preconditions.html#checkState(boolean,%20java.lang.Object))